lib/Dialect/IR/TcpTilingInterfaceImpl.cpp

```diff
@@ -0,0 +1,181 @@
+//===------------------------------------------------------------*- C++ -*-===//
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// Also available under a BSD-style license. See LICENSE.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir-tcp/Dialect/IR/TcpTilingInterfaceImpl.h"
+
+#include "mlir-tcp/Dialect/IR/TcpDialect.h"
+#include "mlir-tcp/Dialect/IR/TcpOps.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Arith/Utils/Utils.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Dialect/Utils/StaticValueUtils.h"
+#include "mlir/IR/DialectRegistry.h"
+#include "mlir/Interfaces/TilingInterface.h"
+
+#include "llvm/Support/Debug.h"
+
+#include <numeric>
+
+using namespace mlir;
+using namespace mlir::tcp;
+
+namespace {
+
+Value getOpFoldResultAsValue(OpFoldResult ofr, OpBuilder &b,
```

Comment:
This and its sibling below already exist upstream. See here:

https://sourcegraph.robot.car/github.robot.car/cruise/mla-robocomp-llvm-project/-/blob/mlir/include/mlir/Dialect/Arith/Utils/Utils.h?L36

Reply:
Thanks for pointing that out. I found some utils related to `OpFoldResult` in https://sourcegraph.com/github.com/llvm/llvm-project@main/-/blob/mlir/include/mlir/Dialect/Utils/StaticValueUtils.h, but didn't realize there are more of these under `arith`. Wonder why they are separate.